### ISSUE TITLE
A UAF crash caused by incorrect use of  |PostTask|

### ISSUE THREAD

**Issue by 0xZOne** [flutter/flutter#95467]
The crash can't be reproduced stably. It took a lot of effort to find the root cause.

```c++
// |PlatformView::Delegate|
void Shell::OnPlatformViewSetNextFrameCallback(const fml::closure& closure) {
  FML_DCHECK(is_setup_);
  FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());

  task_runners_.GetRasterTaskRunner()->PostTask(
      [rasterizer = rasterizer_->GetWeakPtr(), closure = closure]() {
        if (rasterizer) {
          rasterizer->SetNextFrameCallback(std::move(closure));
        }
      });
}
```

<img width="1920" alt="248F1ED2-052D-460E-900F-DB8040339F41" src="https://user-images.githubusercontent.com/26625149/146553571-b88a8bba-3704-4ce1-8b5b-47058d1aea14.png">
<img width="1920" alt="3240BDD1-4C6B-413D-A0A9-89C5A9E58B22" src="https://user-images.githubusercontent.com/26625149/146553659-5f0f77f8-1794-4d92-b3c3-804d41596885.png">

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel master, 2.6.0-12.0.pre.757, on macOS 12.0.1 21A559 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.6.0-12.0.pre.757 at /Users/zero/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 1ba81373ee (4 weeks ago), 2021-11-17 22:01:34 -0500
    • Engine revision a3ef1d5351
    • Dart version 2.16.0 (build 2.16.0-20.0.dev)
    • DevTools version 2.8.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0-rc1)
    • Android SDK at /Users/zero/Library/Android/sdk
    • Platform android-32, build-tools 32.0.0-rc1
    • ANDROID_HOME = /Users/zero/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/zero/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode12.5.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • iPhone (mobile) • 50131fb4dcef9d96405ec953e272bd5c24468370 • ios            • iOS 15.0 19A346
    • Chrome (web)    • chrome                                   • web-javascript • Google Chrome 96.0.4664.110

! Doctor found issues in 1 category.
```

</details>

**Comment by chinmaygarde** [comment]
We took a look at the patch submitted as a fix and I added an explanation for why this is not an issue in the task runner implementation.

We believe this issue is there is [missing weak check on `FlutterViewController` here](https://github.com/flutter/engine/blob/fd74157a3919ef0b0aa0749a23e98fe276ebc569/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L522). The weak pointer gets ferried to the raster thread and then back to the platform thread. By this time, the platform view controller could have been collected. The use of the weak is the right thing to do, but the check was never performed.
